Jint/Runtime/ExceptionHelper.cs

```diff
@@ -14,6 +15,7 @@ public static T ThrowSyntaxError<T>(Engine engine, string message = null)
             return default;
         }
 
+        [DoesNotReturn]
```

Comment:
Saw that in your other PR, perf only? JIT?

Reply:
Basically it helps the compiler. Can write code like:

```csharp
string? thisCanBeNull = SomeMethod();
if (thisCanBeNull is null)
   ThrowHelper.Throw();

// compiler won't complain about possibly being null as it knows the earlier branch won't return this code
thisCanBeNull.Substring(1,2);
```